static/js/settings.js

```diff
@@ -296,6 +326,89 @@ var SettingsScreen = {
     });
   },
 
+  showDiscoverAddonsScreen: function() {
+    this.backButton.href = '/settings/addons';
+    this.addonSettings.classList.remove('hidden');
+    this.addonMainSettings.classList.add('hidden');
+    this.addonDiscoverySettings.classList.remove('hidden');
+
+    const opts = {
+      headers: API.headers(),
+    };
+    fetch('/addons/discover', opts).then(function (response) {
+      return response.json();
+    }).then(function (body) {
+      if (!body) {
+        return;
+      }
+
+      const addonList = document.getElementById('discovered-addons-list');
+      addonList.innerHTML = '';
+
+      for (const addon of body) {
+        const li = document.createElement('li');
```

Comment:
There's nothing fundamentally wrong with this approach, but just note that elsewhere I've abstracted components like this out into a separate object with a view() method to generate an HTML string and a render() method which adds it to the DOM (e.g. thing.js and new-thing.js) - a little bit like React. You then create each item by instantiating an object. It's not particularly sophisticated but it might scale better than putting everything in one giant for loop.

I see that when you moved adapters into settings you removed the Adapter object in adapter.js so maybe we should have a discussion about what approach we want to use for things like this across the front end?

We could even replace the current approach with something more sophisticated like Web Components.

Reply:
I'll split it back out into a view object.